### PR TITLE
Rename upgrade-guard to node-guard

### PR DESCRIPTION
**Pull request by nolancon**

### CODE REVIEW

File: .github/workflows/pre-release.yaml [hunk 1]
```diff
@@ -14,9 +14,9 @@ env:
     RELATED_IMAGE_CSIV1_EXTERNAL_ATTACHER_V3=quay.io/k8scsi/csi-attacher:v3.1.0
     RELATED_IMAGE_CSIV1_EXTERNAL_RESIZER=quay.io/k8scsi/csi-resizer:v1.1.0
     RELATED_IMAGE_STORAGEOS_INIT=storageos/init:v2.1.2
-    RELATED_IMAGE_STORAGEOS_NODE=soegarots/node:v2-release-v2.7.0-24
+    RELATED_IMAGE_STORAGEOS_NODE=storageos/node:v2-release-v2.7.0-24
```
Comment (mhmxs) [inline .github/workflows/pre-release.yaml:17]: This node image points to a pre build on `soegarots`, so this change leads to and image pull issue.

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -15,9 +15,9 @@ env:
     RELATED_IMAGE_CSIV1_EXTERNAL_ATTACHER_V3=quay.io/k8scsi/csi-attacher:v3.1.0
     RELATED_IMAGE_CSIV1_EXTERNAL_RESIZER=quay.io/k8scsi/csi-resizer:v1.1.0
     RELATED_IMAGE_STORAGEOS_INIT=storageos/init:v2.1.2
-    RELATED_IMAGE_STORAGEOS_NODE=soegarots/node:v2-release-v2.7.0-24
+    RELATED_IMAGE_STORAGEOS_NODE=storageos/node:v2-release-v2.7.0-24
```
Comment (mhmxs) [inline .github/workflows/release.yaml:18]: This node image points to a pre build on soegarots, so this change leads to and image pull issue.

File: api/v1/storageoscluster_types.go [hunk 3]
```diff
@@ -180,7 +180,7 @@ type ContainerImages struct {
 	NFSContainer                       string `json:"nfsContainer,omitempty"`
 	APIManagerContainer                string `json:"apiManagerContainer,omitempty"`
 	NodeManagerContainer               string `json:"nodeManagerContainer,omitempty"`
-	UpgradeGuardContainer              string `json:"upgradeGuardContainer,omitempty"`
+	NodeGuardContainer                 string `json:"nodeGuardContainer,omitempty"`
```
Comment (mhmxs) [inline api/v1/storageoscluster_types.go:183]: How we would do migration from old version to new, because this change breaks compatibility?

**Review by mhmxs — APPROVED** [review]

File: api/v1/storageoscluster_types.go (hunk 3 above)
Comment (nolancon) [reply api/v1/storageoscluster_types.go:183]: Good question. I will create a ticket for this as I think it needs some more thought and is not a priority for now

**Comment by mhmxs** [comment]
Tracked here: https://github.com/storageos/operator/pull/206

**Pull request closed without merging**